src/Our.Umbraco.Passless/UI/App_Plugins/UmbracoPassless/BackOffice/overlays/credentials.controller.ts

```diff
@@ -0,0 +1,187 @@
+            close: () => {
+                this.overlayService.close();
+            },
+        });
+    }
+
+    public getCredentials(): void {
+        this.loading = true;
+        this.$http.get(this.getCredentialsEndpoint)
+            .then(response => {
+                this.credentials = response.data;
+                this.loading = false;
+            })
+    }
+
+    public addNewCredentials(): void {
+        this.state = 'adding';
+    }
+
+    public submitRegisterPasslessForm(): void {
+        console.log(`Submitting ${this.registrationAlias}`)
+
+        this.$http.get(`${this.credentialsOptionsEndpoint}?crossPlatform=${this.crossPlatform}`)
+            .then(success => {
+                const makeCredentialsOptions = success.data;
+
+                this.handleUserCredentials(makeCredentialsOptions);
+            }, failure => {
+                console.log(failure);
+                this.notificationsService.error(`failed getting credential options from server`);
+            });
+
+    }
+
+    public close(): void {
+        if (this.$scope.model.close) {
+            this.$scope.model.close();
+        }
+    }
+
+    // TODO: define parameter type
+    private handleUserCredentials(makeCredentialOptions: any) {
+        // Turn the challenge back into the accepted format of padded base64
+        makeCredentialOptions.challenge = coerceToArrayBuffer(makeCredentialOptions.challenge);
+        // Turn ID into a UInt8Array Buffer for some reason
+        makeCredentialOptions.user.id = coerceToArrayBuffer(makeCredentialOptions.user.id);
+
+        makeCredentialOptions.excludeCredentials = makeCredentialOptions.excludeCredentials.map((c: any) => {
+            c.id = coerceToArrayBuffer(c.id);
+            return c;
+        });
+
+        if (makeCredentialOptions.authenticatorSelection.authenticatorAttachment === null) makeCredentialOptions.authenticatorSelection.authenticatorAttachment = undefined;
+
+        navigator.credentials.create({
+            publicKey: makeCredentialOptions
+        }).then(newCredential => {
+            this.prepareNewCredentials(newCredential);
+        }, failure => {
+            console.log(failure);
+            this.notificationsService.error("Failed to register authenticator, possibly due to it already being registered");
+        });
+    }
+
+    // TODO: define parameter type
+    private prepareNewCredentials(newCredentials: any): void {
+        // Move data into Arrays incase it is super long
+        const attestationObject = new Uint8Array(newCredentials.response.attestationObject);
+        const clientDataJSON = new Uint8Array(newCredentials.response.clientDataJSON);
+        const rawId = new Uint8Array(newCredentials.rawId);
+        const base64UrlRawId = coerceToBase64Url(rawId);
+
+        // TODO: is this correct? original code passed the Uint8Array, but btoa requires a string
+        localStorage.setItem("lastCredentials", btoa(newCredentials.rawId));
```

Comment:
yeah it seems it's incorrect - and to me it's also surprising that it worked, I guess the explanation is in the description from MDN

`The btoa() method creates a [Base64](https://developer.mozilla.org/en-US/docs/Glossary/Base64)-encoded ASCII string from a binary string (i.e., a string in which each character in the string is treated as a byte of binary data).` 

but unsure. Anyway It seems it works just passing the ID, instead of the raw ID so i'll change that in my repo :)

Reply:
Upon further investigation i saw this piece of code was some testing that i mistakenly committed! Either with or without the call to btoa the intended behavior would never been achieved. 

I've since fixed it :)

Reply:
Good stuff! Glad I could help :D